doc/i2p.md

```diff
@@ -10,11 +10,20 @@ started with I2P terminology.
 ## Run Bitcoin Core with an I2P router (proxy)
 
 A running I2P router (proxy) with [SAM](https://geti2p.net/en/docs/api/samv3)
-enabled is required (there is an [official one](https://geti2p.net) and
-[a few alternatives](https://en.wikipedia.org/wiki/I2P#Routers)). Notice the IP
-address and port the SAM proxy is listening to; usually, it is
-`127.0.0.1:7656`. Once it is up and running with SAM enabled, use the following
-Bitcoin Core options:
+enabled is required. Options include:
+
+- [i2p-router](https://geti2p.net), the official implementation in Java
+- [i2pd (I2P Daemon)](https://i2pd.readthedocs.io/en/latest), a lighter
+  alternative in C++ (version 2.23 in Debian stable will not work; version 2.35
```

Comment:
I've tested 2.23 with both 22.0rc2 and master and it still manages to work, albeit with a few more warnings, some of which seemed to be false negatives before managing to connect, and the warnings when a peer isn't online are verbose and a bit less helpful:

```
2021-08-06T15:37:05Z I2P: Error connecting to zsxwyo6qcn3chqzwxnseusqgsnuw3maqnztkiypyfxtya4snkoka.b32.i2p:0:
Unexpected reply to "NAMING LOOKUP NAME=zsxwyo6qcn3chqzwxnseusqgsnuw3maqnztkiypyfxtya4snkoka.b32.i2p":
"NAMING REPLY RESULT=INVALID_KEY NAME=zsxwyo6qcn3chqzwxnseusqgsnuw3maqnztkiypyfxtya4snkoka.b32.i2p"
```
Still testing but may just warn here instead.

Reply:
done